### PR TITLE
enhance(client): tune runtime activate for the broken restore

### PR DESCRIPTION
**Pull request by tianweidut**
## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/runtime/model.py [hunk 1]
```diff
@@ -223,6 +224,26 @@ def asdict(self, ignore_keys: t.Optional[t.List[str]] = None) -> t.Dict:
...
+    def mark(cls, status: RuntimeRestoreStatus, rootdir: Path) -> None:
+        ensure_file(rootdir / ".runtime_restore_status", status.value, parents=True)
+
+    @classmethod
+    def get(cls, rootdir: Path) -> RuntimeRestoreStatus:
+        fpath = rootdir / ".runtime_restore_status"
+        if not fpath.exists():
+            return cls.undefined
+        else:
+            return cls(fpath.read_text().strip())
```
Comment (anda-ren) [inline client/starwhale/core/runtime/model.py:244]: what if the content in the fpath is not in the Enum values

File: client/tests/core/test_runtime.py [hunk 2]
```diff
@@ -2512,6 +2513,18 @@ def test_activate(
...
 
+        ensure_file(venv_dir / ".runtime_restore_status", "success", parents=True)
+        StandaloneRuntime.activate(uri=uri, force_restore=False)
+        assert not m_restore.called
+
+        ensure_file(venv_dir / ".runtime_restore_status", "failed", parents=True)
+        StandaloneRuntime.activate(uri=uri, force_restore=False)
+        assert m_restore.called
+
+        ensure_file(venv_dir / ".runtime_restore_status", "restoring", parents=True)
```
Comment (anda-ren) [inline client/tests/core/test_runtime.py:2524]: add one case for broken status file?

Comment (tianweidut) [reply client/tests/core/test_runtime.py:2524]: done

File: client/starwhale/core/runtime/model.py (hunk 1 above)
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py]: add except catch, then make the status undefined.

**Review by anda-ren — APPROVED** [review]
LGTM

**Pull request merged**